### PR TITLE
[WIP/RFC] .parcelignore

### PR DESCRIPTION
**Pull request by DeMoorJasper**
So i gave ``.parcelignore`` a shot because it's been requested in some issues.

Some questions
* Should the ignore, ignore the files completely or include them as is without parsing?
* ~Currently i'm having issues by not wanting to read the ``.parcelignore`` synchronous and seeing that all getDependencies/addDependency logic is synchronous, so if i would change it into an asynchronous function I would change a lot of core logic which is not really a good idea I think. Is there a better way to handle this or should i write ``.parcelignore`` logic to be synchronous?~
* Does this collide with the whole point of 0-config?

Progress:
* [x] Create ignoreChecker following gitignore logic
* [x] ~Find the best way to alter getDependencies~ Should end up in the resolver
* [ ] Ignore files that are flagged by ignoreChecker

related #141 #357 #333

### CODE REVIEW

File: src/utils/checkIgnore.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+const ignore = require('ignore');
+const path = require('path');
+const config = require('./config');
+const fs = require('./fs');
+
+async function getIgnoreFile(location) {
+  let fileLocation = await config.resolve(location, ['.parcelignore']);
+  let ignoreFile = '';
+  if (fileLocation) ignoreFile = (await fs.readFile(fileLocation)).toString();
```
Comment (sabeurthabti) [inline src/utils/checkIgnore.js]: maybe some sort of error handling here? (`try/catch`)

**Comment by devongovett** [comment]
Good questions! Also related: #474. Not sure I have the answers yet. I think we need a bit more input from others on how this should work.

On your second point, I don't think I'd do this in `addDependency`. I think it's ok if the dependency is still parsed. Instead, you could check in the Resolver, and return null if the file is ignored. That will cause a few other changes to check for null in Bundler, but should be easier than trying to change the logic in every asset type.

**Comment by thejameskyle** [comment]
I would like to discuss this in the context of a more complete (`.parcelrc`) config format. I think it's possible that this would not be necessary.

**Comment by mikkiek** [comment]
Hello,

Any ideas when can i expect parcelignore "miracle"?

**Comment by agauniyal** [comment]
In my case(#357) I wanted to ignore certain sections, not the whole file itself 😅, would that be possible with this PR?

**Comment by mathiasrw** [comment]
A central use case would be to link from your HTML to a legacy entry point. 

Let's say you have a nice static website up and running with parceljs but you would like to link to your crazy old PHP3 system located at `/login` - at the current version `<a href="/login">login</a>` will result in an error because it cant find `/../../../../../login` (at the root of the drive)

I suggest adding `/login` to the `.parcelignore` would ignore it as a dependency and just move along without any output.

**Comment by DeMoorJasper** [comment]
@mathiasrw I just closed it due to inactivity if parcelignore ever gets approved i’ll just make a new pr

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by dinvlad** [comment]
+1 on special URLs used by Firebase

**Comment by goyney** [comment]
Would like to see this reopened. Here's a PERFECT use case for it: https://github.com/moment/moment/issues/2517#issuecomment-393704231

**Comment by DeMoorJasper** [comment]
Please do not comment on an old closed PR, open up a new issue instead.

**Pull request closed without merging**